net/coap/client.go

```diff
@@ -491,8 +491,11 @@ func DialUDP(ctx context.Context, addr string, opts ...DialOptionFunc) (*ClientC
 		cfg = o(cfg)
 	}
 	dopts := make([]udp.DialOption, 0, 4)
-	if cfg.KeepAlive != nil {
-		dopts = append(dopts, udp.WithKeepAlive(cfg.KeepAlive))
+	if cfg.KeepaliveTimeout != 0 {
+		dopts = append(dopts, udp.WithKeepAlive(3, cfg.KeepaliveTimeout/3, func(cc inactivity.ClientConn) {
+			cc.Close()
+			cfg.errors(fmt.Errorf("keep alive was reached fail limit:: closing connection"))
```

Comment:
I proposed new config for CoAP Gateway keepalive mechanism. 
```
apis:
  coap:
    address: ""
    externalAddress: ""
    timeout: 0s 
    goroutineSocketHeartbeat: 4s
    capabilities:
      maxMessageSize: 1024
      keepalive:
        enabled: true
        timeout: 5s
        maxRetries: 6
      blockwiseTransfer:
        enabled: true
        blockSize: 1024
```

Reply:
I don't want to change it because it is used by sdk, not by coap-gw so the configuration will be stored at coap-gateway.